File: 1899 B. 250 Thousand Tons of TNT/Solution.py
# 315532662	        Apr/14/2025 11:08UTC+6	        Krishno_MD	        1899B - 250 Thousand Tons of TNT	        PyPy 3-64	        Accepted	        264 ms	        20600 KB

# Krishno_MD_KUET_CSE

t = int(input())

for i in range(t):
    n = int(input())
    v = list(map(int,input().split()))
    div = []
    i=1
    while i*i<=n:
        if n%i==0:
            div.append(i)
            if i != n/i:
                div.append(n/i)
        i+=1
    a=0
    for num in div:
        maxi,mini,cnt,su = 0, float('inf'),0,0
        for i in v:
            su += i
            cnt+=1
            if cnt==num:
                maxi=max(maxi,su)
                mini=min(mini,su)
                cnt=0
                su=0
        a=max(a,maxi-mini)
    print(a)

File: 1899 B. 250 Thousand Tons of TNT/Solution.cpp
// 315532571	        Apr/14/2025 11:07UTC+6	        Krishno_MD	        1899B - 250 Thousand Tons of TNT	        C++20 (GCC 13-64)	        Accepted	        265 ms	        0 KB

#include<bits/stdc++.h>
using namespace std; 
#define ll long long

// Krishno_MD_KUET_CSE

void solve(){
    int n; cin>>n;
    vector<int> v(n);
    for(int i=0; i<n; i++){cin>>v[i];}
    vector<int> div;
    for(int i=1; i*i<=n; i++){
        if(n%i==0){
            div.push_back(i);
            if(i != n/i){
                div.push_back(n/i);
            }
        }
    }
    ll a=0;
    for(int num : div){
        ll maxi = 0, mini = LLONG_MAX;
        int cnt =0; long long su=0;
        for(int i=0; i<n; i++){
            su += v[i]; cnt++;
            if(cnt==num){
                maxi = max(maxi,su);
                mini = min(mini,su);
                cnt=0; su=0;
            }
        }
        a = max(a,maxi-mini);
    }
    cout<<a<<endl;
}

int main(){
    int t;
    cin >> t;
    while(t--){
        solve();
    }
    return 0;
}
